samples/vboxwrapper/vbox_vboxmanage.cpp

```diff
@@ -883,73 +878,91 @@ int VBOX_VM::poll(bool log_state) {
             //
             // So for now, go with what VboxManage is reporting.
             //
-            if (vmstate == "running") {
+            if (vmstate == "Running") {
                 online = true;
                 saving = false;
                 restoring = false;
                 suspended = false;
                 crashed = false;
-            } else if (vmstate == "paused") {
+            }
+            else if (vmstate == "Paused") {
                 online = true;
                 saving = false;
                 restoring = false;
                 suspended = true;
                 crashed = false;
-            } else if (vmstate == "starting") {
+            }
+            else if (vmstate == "Saved") {
+                online = false;
+                saving = false;
+                restoring = false;
+                suspended = true;
+                crashed = false;
+            }
+            else if (vmstate == "Starting") {
                 online = true;
                 saving = false;
                 restoring = false;
                 suspended = false;
                 crashed = false;
-            } else if (vmstate == "stopping") {
+            }
+            else if (vmstate == "Stopping") {
                 online = true;
                 saving = false;
                 restoring = false;
                 suspended = false;
                 crashed = false;
-            } else if (vmstate == "saving") {
+            }
+            else if (vmstate == "Saving") {
                 online = true;
                 saving = true;
                 restoring = false;
                 suspended = false;
                 crashed = false;
-            } else if (vmstate == "restoring") {
+            }
+            else if (vmstate == "Restoring") {
                 online = true;
                 saving = false;
                 restoring = true;
                 suspended = false;
                 crashed = false;
-            } else if (vmstate == "livesnapshotting") {
+            }
+            else if (vmstate == "LiveSnapshotting") {
                 online = true;
                 saving = false;
                 restoring = false;
                 suspended = false;
                 crashed = false;
-            } else if (vmstate == "deletingsnapshotlive") {
+            }
+            else if (vmstate == "DeletingsnapshotOnline") {
```

Comment:
The old and new string differ in more than lower/uppercase shouldn't we be backwards compatible with older VirtualBox version? If not the wrapper should have a minimum requirement that is checked at startup. The same is true for some of the following state names.

Reply:
If we go with this approach we would not depend on the showvminfo command and thus as long as VirtualBox does not change it's MachineState names we would not need to change these strings.

Reply:
My point is that both versions (showvminfo and VBox.log output) should work here. Something like:
```
 vmstate =strtolower( read_vm_log() );
...
else if (vmstate == "deletingsnapshotonline" || vmstate="deletingsnapshotlive") {
...
}
```
This way we can reuse this part of the code for the old showvminfo output as well as the new VBox.log output